clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10406,6 +10406,42 @@ void OffloadDeps::ConstructJobMultipleOutputs(Compilation &C,
   constructJob(C, JA, Outputs, Inputs, TCArgs, LinkingOutput);
 }
 
+// Utility function to gather all options for LLVM to SPIR-V translation using
+// SPIR-V backend.
+static void getSPIRVBackendOpts(const llvm::opt::ArgList &TCArgs,
+                                ArgStringList &TranslatorArgs) {
+  TranslatorArgs.push_back(TCArgs.MakeArgString("-filetype=obj"));
+  TranslatorArgs.push_back(
+      TCArgs.MakeArgString("-mtriple=spirv64-unknown-unknown"));
+  TranslatorArgs.push_back(TCArgs.MakeArgString("-O0"));
+  TranslatorArgs.push_back(
+      TCArgs.MakeArgString("--avoid-spirv-capabilities=Shader"));
+  TranslatorArgs.push_back(
+      TCArgs.MakeArgString("--translator-compatibility-mode"));
+
+  // Disable all the extensions by default
+  std::string ExtArg("-spirv-ext=");
+  std::string DefaultExtArg =
+      "+SPV_EXT_shader_atomic_float_add,+SPV_EXT_shader_atomic_float_min_max"
+      ",+SPV_KHR_no_integer_wrap_decoration,+SPV_KHR_float_controls"
+      ",+SPV_KHR_expect_assume,+SPV_KHR_linkonce_odr";
+  std::string INTELExtArg = ",+SPV_INTEL_subgroups,+SPV_INTEL_function_pointers"
+                            ",+SPV_INTEL_arbitrary_precision_integers"
+                            ",+SPV_INTEL_variable_length_array";
+  ExtArg = ExtArg + DefaultExtArg + INTELExtArg;
+
+  // Other args
+  ExtArg += ",+SPV_INTEL_bfloat16_conversion"
```

Comment:
Do we have an idea of how many extensions that we usually use for llvm-spirv but are not supported by the spv backend?

Reply:
This is being tracked internally. I think we are at around 50%. @VyacheslavLevytskyy for a better answer.

Thanks

Reply:
I don't track exact number, so 50% as good an estimation as any other. We have different degree of support of CPU, GPU and FPGA at the moment, total 23 implemented (with an important recent addition of SPV_KHR_cooperative_matrix), and very roughly less than 10 to add as a short-term goal.